### DISCUSSION TITLE
MappingFunction working for one sensor and not another

### DISCUSSION THREAD

**Discussion by mattibatti** [tttapa/Control-Surface#1031] [General]
I am trying to map the input from a softpot and a tilt sensor to control changes using the Potentioeter.map() example. The mapping is working great for the softpot, but the tilt sensor is only transmitting changes from 35-45, instead of 0-127. I have tried using multiple different values in the mapping function but nothing seems to work. Any ideas? Thanks so much.

```
#include <Control_Surface.h>

USBMIDI_Interface midi;

CCPotentiometer softpot {A0, 50};
CCPotentiometer tilt {A1, 51};

analog_t minimum_values[] {0, 260};
analog_t maximum_values[] {7000, 390} ;

template <size_t I>
analog_t mappingFunction(analog_t raw) {
    raw = constrain(raw, minimum_values[I], maximum_values[I]);
    return map(raw, minimum_values[I], maximum_values[I], 0, 16383);
}

void setup() {
  softpot.map(mappingFunction<0>);
  tilt.map(mappingFunction<1>);
  Control_Surface.begin();
}

void loop() {
  Control_Surface.loop();
}
```

**Comment by mattibatti** [comment]
Hi following up to see if anyone has an idea of why I am still only getting limited range output from my sensor? I have tried every thing I can think of :/

**Comment by tttapa** [comment]
Please provide some more debugging information:

```cpp
template <size_t I>
analog_t mappingFunction(analog_t raw) {
    analog_t constr = constrain(raw, minimum_values[I], maximum_values[I]);
    analog_t ret = map(constr, minimum_values[I], maximum_values[I], 0, 16383);
    if (I == 1)
        Serial << raw << '\t' << constr << '\t' << ret << '\n';
}

void setup() {
  Serial.begin(115200); 
  // ...
}
```
Keep in mind that your ADC is only 10 bits, whereas the raw values are converted to 14 bits using `AH::increaseBitDepth<14, 10, analog_t>(analogRead(pin))`. To determine your bounds, consider using `tilt.getRawValue()`, and print that to the serial monitor.

**Reply by mattibatti** [reply]
Thanks so much for your reply, I am embarrased to say it was just a naming issue with the variables. I used your debugging method and to confirm that the problem wasn't with the mapping function, so I looked all around and found the naming issue. Now it is working! Thanks again for all your help, m
